OsmAnd-java/src/net/osmand/util/GeoPointParserUtil.java

```diff
@@ -813,15 +844,57 @@ public boolean isGeoPoint() {
 			return geoPoint;
 		}
 
+		private String formatDouble(double d) {
+			if(d == (long) d)
+				return String.format(Locale.ENGLISH, "%d", (long)d);
+			else
+				return String.format("%s", d);
+		}
+
 		public boolean isGeoAddress() {
 			return geoAddress;
 		}
 
+		/**
+		 * Generates a URI string according to https://tools.ietf.org/html/rfc5870 and
+		 * https://developer.android.com/guide/components/intents-common.html#Maps
+		 */
 		@Override
 		public String toString() {
```

Comment:
I just following the patterns established by Java SE and Android: classes that
represent URIs all use `toString()` to output the complete URI:

* http://docs.oracle.com/javase/7/docs/api/java/net/URL.html#toString()
* http://docs.oracle.com/javase/7/docs/api/java/net/URI.html#toString()
* https://developer.android.com/reference/android/net/Uri.html#toString()

And classes that "build" data classes have a `toString()` method that outputs
a debug string:

* https://developer.android.com/reference/android/net/Uri.Builder.html#toString()

`GeoPointParserUtil` is the "builder", and `GeoParsedPoint` represents data
only, so is like java.net.URL, java.net.URI, and android.net.Uri.